api/src/test/java/org/motechproject/nms/api/osgi/MobileAcademyControllerBundleIT.java

```diff
@@ -0,0 +1,80 @@
+package org.motechproject.nms.api.osgi;
+
+import org.apache.commons.httpclient.HttpStatus;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.codehaus.jackson.map.deser.ValueInstantiators;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.motechproject.nms.api.utils.RequestBuilder;
+import org.motechproject.nms.api.web.BaseController;
+import org.motechproject.nms.api.web.contract.mobileAcademy.SaveBookmarkRequest;
+import org.motechproject.nms.mobileacademy.service.MobileAcademyService;
+import org.motechproject.testing.osgi.BasePaxIT;
+import org.motechproject.testing.osgi.container.MotechNativeTestContainerFactory;
+import org.motechproject.testing.osgi.http.SimpleHttpClient;
+import org.motechproject.testing.utils.TestContext;
+import org.ops4j.pax.exam.ExamFactory;
+import org.ops4j.pax.exam.junit.PaxExam;
+import org.ops4j.pax.exam.spi.reactors.ExamReactorStrategy;
+import org.ops4j.pax.exam.spi.reactors.PerSuite;
+
+import javax.inject.Inject;
+
+import java.io.IOException;
+
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Integration tests for mobile academy controller
+ */
+@RunWith(PaxExam.class)
+@ExamReactorStrategy(PerSuite.class)
+@ExamFactory(MotechNativeTestContainerFactory.class)
+public class MobileAcademyControllerBundleIT extends BasePaxIT {
+
+    @Inject
+    private MobileAcademyService mobileAcademyService;
+
+    @Test
+    public void testBookmarkBadCallingNumber() throws IOException, InterruptedException {
+
+        String endpoint = String.format("http://localhost:%d/api/mobileacademy/bookmarkWithScore",
+                TestContext.getJettyPort());
+
+        HttpPost request = RequestBuilder.createPostRequest(endpoint, new SaveBookmarkRequest());
+        assertTrue(SimpleHttpClient.execHttpRequest(request, HttpStatus.SC_BAD_REQUEST, RequestBuilder.ADMIN_USERNAME, RequestBuilder.ADMIN_PASSWORD));
+    }
+
+    @Test
+    public void testBookmarkBadCallId() throws IOException, InterruptedException {
+
+        String endpoint = String.format("http://localhost:%d/api/mobileacademy/bookmarkWithScore",
+                TestContext.getJettyPort());
+        SaveBookmarkRequest bookmarkRequest = new SaveBookmarkRequest();
+        bookmarkRequest.setCallId(BaseController.SMALLEST_15_DIGIT_NUMBER - 1);
+        HttpPost request = RequestBuilder.createPostRequest(endpoint, bookmarkRequest);
+        assertTrue(SimpleHttpClient.execHttpRequest(request, HttpStatus.SC_BAD_REQUEST, RequestBuilder.ADMIN_USERNAME, RequestBuilder.ADMIN_PASSWORD));
+    }
+
+    @Test
+    public void testSetValidBookmark() throws IOException, InterruptedException {
+
```

Comment:
Yes, you are right about the test name. Updated it and added a positive test as well.

And yes for the bad response validation.. but wanted to get something in before I dove into more IT and UT validations. Next checkin?